astropy/coordinates/angles.py

```diff
@@ -85,54 +85,43 @@ class Angle(u.Quantity):
     _include_easy_conversion_members = True
 
     def __new__(cls, angle, unit=None, dtype=None, copy=True):
-        unit = cls._convert_unit_to_angle_unit(unit)
-        if (unit is not None and not unit.is_equivalent(u.radian)):
-            raise u.UnitsError("Requested unit {0} is not convertible to an "
-                               "angle".format(unit))
 
-        if isinstance(angle, u.Quantity):
-            # This includes Angle subclasses as well
+        if not isinstance(angle, u.Quantity):
             if unit is not None:
-                angle = angle.to(unit).value
-            else:
-                unit = angle.unit
                 unit = cls._convert_unit_to_angle_unit(unit)
-                if not unit.is_equivalent(u.radian):
-                    raise u.UnitsError(
```

Comment:
The file is fully covered except for 6 lines for default cases/exceptions in `to_string`. I'd rather not go there in this PR... But I did add an explicit test for inputting something with a non-angular unit.